src/mono/wasm/runtime/hybrid-globalization/change-case.ts

```diff
@@ -31,20 +27,19 @@ export function mono_wasm_change_case_invariant(src: number, srcLength: number,
     }
 }
 
-export function mono_wasm_change_case(culture: MonoStringRef, src: number, srcLength: number, dst: number, destLength: number, toUpper: number, is_exception: Int32Ptr, ex_address: MonoObjectRef) : void{
+export function mono_wasm_change_case(culture: MonoStringRef, src: number, srcLength: number, dst: number, dstLength: number, toUpper: number, is_exception: Int32Ptr, ex_address: MonoObjectRef): void {
     const cultureRoot = mono_wasm_new_external_root<MonoString>(culture),
         exceptionRoot = mono_wasm_new_external_root<MonoObject>(ex_address);
-    try{
-        const cultureName = conv_string_root(cultureRoot);
+    try {
+        const cultureName = monoStringToString(cultureRoot);
         if (!cultureName)
             throw new Error("Cannot change case, the culture name is null.");
-        const input = get_utf16_string(src, srcLength);
+        const input = utf16ToString(src, src + 2 * srcLength);
```

Comment:
This place used to use loop-only. When we change it to using the TextDecoder on browser, we might have some hybrid tests failing - encoders/decoders behave like NLS, not like ICU.